tests/FrontendSpec.hs

```diff
@@ -1,78 +1,37 @@
-{-# LANGUAGE DataKinds             #-}
-{-# LANGUAGE FlexibleContexts      #-}
-{-# LANGUAGE FlexibleInstances     #-}
-{-# LANGUAGE OverloadedStrings     #-}
-{-# LANGUAGE ScopedTypeVariables   #-}
-{-# LANGUAGE TypeFamilies          #-}
-{-# LANGUAGE TypeOperators         #-}
+module FrontendSpec where
 
-{-# OPTIONS_GHC -Werror -Wall #-}
-
-module FrontendSpec
-where
-
-import Data.String.Conversions (ST, cs)
-import Lucid (renderText, toHtml)
-import Test.Hspec (Spec, describe, context, it)
-import Test.Hspec.Wai (with, get, shouldRespondWith, pending, ResponseMatcher(ResponseMatcher))
-import Servant
-import Network.Wai
-
-import Action
-import Action.Implementation
-import Config
-import Frontend
-import Frontend.Core
-import qualified Persistent.Implementation.STM
+import AulaTests
 
 spec :: Spec
 spec = do
-    describe "catchAulaExcept" . with testAppAulaExcept $ do
-        let handles500 = do
-                it "writes error message to stdout" $ do
-                    pending
-                it "sends an apologetic 500 http response" $ do
-                    pending
-                    get "/error" `shouldRespondWith` 500
-                it "bases http response on type `Page500`" $ do
-                    pending
+    describe "servant exception handling" . around withServer $ do
+        context "on `undefined`" $ do
+            it "writes error message to stdout" $ \_query -> do
+                pending
+            it "sends an apologetic 500 http response" $ \query -> do
+                get query "/testing/undefined" `shouldRespond` [codeShouldBe 500]
+            it "bases http response on type `Page500`" $ \_query -> do
```

Comment:
This may actually be the reason for keeping `catchAulaExcept`.  I will leave a comment in the code for now, and we can revisit the problem later.  (500 errors shouldn't happen in the first place.)